### PR TITLE
more perf updates

### PR DESCRIPTION
**Pull request by mcollina**
cc @jasnell

### CODE REVIEW

File: lib/internal/http2.js [hunk 1]
```diff
@@ -1605,20 +1606,7 @@ class Http2ClientRequest extends Http2Outgoing {
...
-              if (Array.isArray(value) && value.length > 0) {
-                for (const item of value)
-                  this.stream.addTrailer(key, String(item));
-              } else {
-                this.stream.addTrailer(key, String(value));
-              }
-            }
-          }
-        });
+        this.stream.once('handle', kBeginSendHandle(this))
```
Comment (targos) [inline lib/internal/http2.js]: Do you know why perf is better with this change?

File: lib/internal/http2.js [hunk 2]
```diff
@@ -1629,6 +1617,23 @@ class Http2ClientRequest extends Http2Outgoing {
   }
 }
 
+function kBeginSendHandle(that) {
```
Comment (targos) [inline lib/internal/http2.js]: don't we use the prefix `k` for constants?

File: lib/internal/http2.js (hunk 1 above)
Comment (mcollina) [reply lib/internal/http2.js]: the function declaration is hoisted. It happens even if the if statement is false.

File: lib/internal/http2.js (hunk 2 above)
Comment (mcollina) [reply lib/internal/http2.js]: I just called it in a way that will show where it is used. I'll drop the k.

**Comment by mcollina** [comment]
@targos check it out now. I've removed all the closure creation, and the for loop can be optimized by V8 (short-lived closures are currently not optimized).

**Comment by targos** [comment]
Nice! LGTM

**Review by jasnell — APPROVED** [review]

**Comment by jasnell** [comment]
Landed via https://github.com/nodejs/http2/pull/23

**Pull request closed without merging**